ansible_base/authentication/serializers/authenticator_map.py

```diff
@@ -17,17 +18,42 @@ def validate(self, data) -> dict:
         map_type = data.get('map_type', None)
         team = data.get('team', None)
         org = data.get('organization', None)
+        role = data.get('role', None)
+
         if map_type == 'team' and (not team or team == ''):
             errors["team"] = "You must specify a team with the selected map type"
-        if map_type == 'team' and (not org or org == ''):
-            errors["organization"] = "You must specify an organization with the selected map type"
-        if map_type == 'organization' and (not org or org == ''):
+        if map_type in ['team', 'organization'] and (not org or org == ''):
             errors["organization"] = "You must specify an organization with the selected map type"
+        if map_type in ['team', 'organization', 'role'] and role is None:
+            errors["role"] = "You must specify a role with the selected map type"
+        if map_type in ['allow', 'is_superuser'] and role is not None:
+            errors["role"] = "You cannot specify role with the selected map type"
+
+        if role is not None:
+            errors.update(self.validate_role_data(role, org, team))
```

Comment:
This is an interesting problem because above we are validating the map types and here we are validating the roles. This function is role centric, looking at the role and then validating the inputs.
But I think we are missing checks for "does this role make sense for the map type". If I am reading things right and I selected an organization map type but passed in a Team and a Team type role I think I would get through this validation logic.

So we likely need an additional check somewhere to say "if you selected org map type and your role is not an org map type raise 'For an organization map type you must select an organization based role'".